### DISCUSSION TITLE
Open in browser

### DISCUSSION THREAD

**Discussion by jtagcat** [drakkan/sftpgo#1309] [Ideas]
Feature request: Preview files in the browser when clicking on it in a list.

This can be done by using `Content-Disposition: inline;` instead of `Content-Disposition: attachment;`.

The feature could include a file extension whitelist, but doesn't have to, since the browser downloads files it can't display anyway.

Files can still be directly downloaded from the menu.

**Comment by drakkan** [comment]
Hello, 

thank you for your suggestion.

What's wrong with the preview icon?

![Schermata del 2023-05-30 12-50-52](https://github.com/drakkan/sftpgo/assets/553263/3dc24413-97b7-4022-b4d7-9555a2bd7178)

Do you have trouble with a particular type of file that the browser can display and SFTPGo doesn't display the icon for preview?

**Reply by jtagcat** [reply]
I don't see the preview icon anywhere: pdf, mp3, jpg, flac files.

In my opinion, if I click on the filename, it should open a preview, the download being the secondary action instead.

**Reply by jtagcat** [reply]
The preview icon is not available for shared links.

**Comment by alexitx** [comment]
Please make an option, at least for public shares. It will be really useful. Currently, images, video, documents and other files readable by browsers cannot be previewed and instead begin to download. Proposal - a setting, such as `Direct download: true/false` when creating a share, or available globally which will disregard file extensions and always set the `Content-Disposition: inline` header for all files.

Edit: after a quick look, I see a function [`downloadFromShare`](https://github.com/drakkan/sftpgo/blob/main/internal/httpd/api_shares.go#L274), which I assume handles downloading files from public shares, but ignores the `inline` parameter, and a different function [`downloadBrowsableSharedFile`](https://github.com/drakkan/sftpgo/blob/main/internal/httpd/api_shares.go#L224), which seems to be used only in the REST API, but checks for the `inline` parameter and, if present, doesn't set `Content-Disposition: attachment`. Can this be done in a similar way for downloading from the web client instead of the API?

**Reply by drakkan** [reply]
please try sending a PR. We will discuss further implementation details once we have a working (and XSS safe) PR. Thank you

**Reply by alexitx** [reply]
I would love to, but I have no experience in Go. If someone decides to open a PR, I can help with testing.
